rust/kernel/traits.rs

```diff
@@ -8,7 +8,7 @@ use alloc::{alloc::AllocError, sync::Arc};
 
 /// Trait which provides a fallible version of `pin()` for pointer types.
 ///
-/// Common pointer types which implement a `pin()` method include [`Box`], [`Arc`] and [`Rc`].
+/// Common pointer types which implement a `pin()` method include [`alloc::boxed::Box`] and [`Arc`].
```

Comment:
Maybe keep `Box` as visible name but use `alloc::boxed::Box` as link target?

Reply:
Agreed, it would look nicer.

Reply:
How does that work? Any suggestions?

Reply:
Luckily I guessed correctly :)